buildtools/smoketests.proj

```diff
@@ -0,0 +1,90 @@
+<Project ToolsVersion="Current" DefaultTargets="smoke-tests" xmlns="http://schemas.microsoft.com/developer/msbuild/2003"> 
+    <PropertyGroup>
+        <RootPath Condition="'$(RootPath)'==''">$(MSBuildProjectDirectory)</RootPath>
+
+        <!-- We need to skip recipes that use Docker when running on Windows CodeBuild -->
+        <ShouldRunDockerTests>true</ShouldRunDockerTests>
+        <ShouldRunDockerTests Condition="'$(OS)' == 'Windows_NT' AND '$(CODEBUILD_BUILD_ID)' != ''">false</ShouldRunDockerTests>
+
+        <!-- Generate a suffix for CloudFormation stack names, in case multiple smoke tests are running at once -->
+        <SafeOperatingSystem>$(OS)</SafeOperatingSystem>
```

Comment:
nit: could you specify an inverse condition here? Its easy to mess this up if in the future someone mistakenly reorders them.

Reply:
I'm worried about multiple runs of the canary, failed deployments and failing to delete some stacks. It is possible to get to a state where we have stacks that linger to the next canary run (although unlikely). Some of our tests reuse the same names and IDs. I would feel better if we can use a more unique suffix like a guid instead of the OS name. Though I don't want us to over complicate things. I'm fine moving forward and seeing what the canaries will be like in action.

Reply:
The full suffix includes `$([System.DateTimeOffset]::UtcNow.ToUnixTimeSeconds()` below down on L12.

So each run of the canary will use unique stack names based on the timestamp. If the Windows and Linux runners start at the same second, then the `SafeOperatingSystem` will disambiguate them.

Reply:
I remember deleting that comment. Dunno why it posted. I deleted it when I read a few lines later where you were doing `$([System.DateTimeOffset]::UtcNow.ToUnixTimeSeconds()`